ipn/localapi/localapi.go

```diff
@@ -67,11 +83,30 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		h.serveGoroutines(w, r)
 	case "/localapi/v0/status":
 		h.serveStatus(w, r)
+	case "/localapi/v0/bugreport":
+		h.serveBugReport(w, r)
 	default:
 		io.WriteString(w, "tailscaled\n")
 	}
 }
 
+func (h *Handler) serveBugReport(w http.ResponseWriter, r *http.Request) {
+	// Require write access out of paranoia that the bugreport
+	// might contain something sensitive.
+	if !h.PermitWrite {
+		http.Error(w, "bugreport access denied", http.StatusForbidden)
+		return
+	}
+
+	logMarker := fmt.Sprintf("BUG-%v-%v-%v", h.backendLogID, time.Now().UTC().Format("UTC20060102150405Z"), randHex(8))
+	if v := r.FormValue("note"); v != "" {
+		logMarker += "-" + v
```

Comment:
url.QueryEscape the note perhaps?

Reply:
I changed this a bit, lmk what you think.
I removed the suffix and just logged the entire note in a separate line.